src/test/resources/org/rust/ide/inspections/fixtures/unnecessary_parenthesis.rs

```diff
@@ -0,0 +1,9 @@
+fn main() {
+    let x = 5;
+
+    if <warning descr="Unnecessary parenthesis">(x == 5)</warning> {
+        println!("x is five!");
+    } else {
+        println!("x is not five :(");
+    }
+}
```

Comment:
It might be nice to add a test case for the @matklad's 
```rust
if (true) or (false) {}
```
Of course, no warning should be triggered.